docs/connectors/postgresql.asciidoc

```diff
@@ -2,5 +2,1044 @@
+* `txId` represents the identifier of the server transaction which caused the event
+* `ts_usec` represents the number of microseconds since Unix Epoch as the server time at which the transaction was committed
+
+[[events]]
+=== Events
+
+All data change events produced by the PostgreSQL connector have a key and a value, although the structure of the key and value depend on the table from which the change events originated (see link:#topic-names[Topic names]).
+
+[NOTE]
+====
+Starting with Kafka 0.10, Kafka can optionally record with the message key and value the http://kafka.apache.org/documentation.html#upgrade_10_performance_impact[_timestamp_] at which the message was created (recorded by the producer) or written to the log by Kafka.
+====
+
+[WARNING]
+====
+The Debezium PostgreSQL connector ensures that all Kafka Connect _schema names_ are http://avro.apache.org/docs/current/spec.html#names[valid Avro schema names]. This means that the logical server name must start with Latin letters or an underscore (e.g., [a-z,A-Z,\_]), and the remaining characters in the logical server name and all characters in the schema and table names must be Latin letters, digits, or an underscore (e.g., [a-z,A-Z,0-9,\_]). If not, then all invalid characters will automatically be replaced with an underscore character.
+
+This can lead to unexpected conflicts when the logical server name, schema names, and table names contain other characters, and the only distinguishing characters between table full names are invalid and thus replaced with underscores.
+====
+
+Debezium and Kafka Connect are designed around _continuous streams of event messages_, and the structure of these events may change over time. This could be difficult for consumers to deal with, so to make it very easy Kafka Connect makes each event self-contained. Every message key and value has two parts: a _schema_ and _payload_. The schema describes the structure of the payload, while the payload contains the actual data.
+
+[[change-events-key]]
+==== Change event's key
+
+For a given table, the change event's key will have a structure that contains a field for each column in the primary key (or unique key constraint) of the table at the time the event was created. 
+
+Consider a `customers` table defined in the `public` database schema:
+
+[source,sql,indent=0]
+----
+CREATE TABLE customers (
+  id SERIAL,
+  first_name VARCHAR(255) NOT NULL,
+  last_name VARCHAR(255) NOT NULL,
+  email VARCHAR(255) NOT NULL,
+  PRIMARY KEY(id)
+);
+----
+
+If the `database.server.name` configuration property has the value `PostgreSQL-server-1`, every change event for the `customers` table while it has this definition will feature the same key structure, which in JSON looks like this:
+
+[source,json,indent=0]
+----
+  {
+    "schema": {
+      "type": "struct",
+      "name": "PostgreSQL-server-1.public.customers.Key"
```

Comment:
This doesn't comply with Avro Schema names, does it? Won't the `-` characters be replaced with `_`? If so, wouldn't it be easier/simpler to have the example where everything matches?